### PR TITLE
Disease antagonist

### PR DESCRIPTION
**Pull request by Cruix**
:cl: 
add: Added a new mini antagonist, the sentient disease.
/:cl:

Closes #35954

Added a new mini antagonist, a disease that evolves new symptoms and powers. It gains points for infecting multiple people at the same time, and it can use those points to purchase symptoms. The overmind can jump between all infected hosts.

Right now the only non-symptom abilities are one to make the host cough, and one to make the host sneeze. The cough will try to infect people nearby the host, while the sneeze has a longer range, but can only infect people in front of the host. The cough and sneeze symptoms are different, can be purchased separately, and act like they do for regular diseases.

Evolution menu:
menu

Icons for the evolution menu, cough, and sneeze buttons:
buttons

Overmind and hud sprites:
hud

Roundend report:
report

### CODE REVIEW

**Comment by Ordonis** [comment]
More disease functionality and use for virologist is good

**Comment by Buggy123** [comment]
we plague inc now

**Comment by fludd12** [comment]
This is awesome!

**Comment by MoreRobustThanYou** [comment]
Add sentient disease only symptoms

**Comment by Buggy123** [comment]
Given how there's beneficial symptoms on that list, are viruses gonna have the choice to go for super deadly and virulent or helpful but slow spreading, etc? It'd be interesting, though I hope that that wouldn't end up with every single virus becoming beneficial every time.

**Comment by fludd12** [comment]
Maybe sentient viruses could spend points to 'unlock' a symptom, then have the ability to turn it on/off? Then, for a higher cost, you could unlock the ability to control who gets what symptoms. Reward people for spreading you, and be a detriment to those who are trying to cure you!

**Comment by vuonojenmustaturska** [comment]
Oo, this looks nice. 

Idea: maybe have "upgraded" coughing/vomit symptoms where they cause smoke/foam to appear? We have a long-lasting foam subtype courtesy of goof that could be good for this.

**Comment by WJohn** [comment]
Sprites are acceptable, though seriously why on earth do you play with stretch to fit do you enjoy suffering? Get a bigger monitor, it makes a world of difference when you're not playing at 1.5x scale.

**Review by WJohn — DISMISSED** [review]

**Comment by Ratvar** [comment]
Maybe reward virus for hurting/killing hosts, so it's not "beneficial virus but observer" ?
Also, medics scanning people at event mark to kill them/suicide out of spite.
"Hey dude is Jack Brens an antag cough me once if yes if you are here"

**Comment by ExcessiveUseOfCobblestone** [comment]
This is awesome.

However, If the objective is to survive, they're always going to be epic heal disease to at least 1 individual for greentext.

**Comment by Razharas** [comment]
It should evolve up to zombie virus including the ability to take control over dead bodies of infected people

**Comment by Cruix** [comment]
You can click on the name of a symptom or ability in the menu to get more information on it, including the stat changes:
untitled

People trying to meta with the virus is no different than any other ghost role, and if medics want to start scanning people for disease just to fuck over the virus, then they are doing the job they are supposed to do.

Becoming purely beneficial virus will take considerable time, since all the healing symptoms require a large number of infected to unlock.

There is also a two minute cooldown to buying or selling abilities, so you cannot just change your whole disease on a whim.

I would like to get this testmerged and fix the inevitable deluge of bugs before making any new powers or symptoms. Zombies are a definite possibility, as are individual symptoms, with the bonus of a "kill the viro" objective.

**Comment by Spaceinaba** [comment]
\>kill the viro objective
hahaha I love it

**Comment by Ordonis** [comment]
Testmerge when, plague when, handwashing when

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Will need further review

**Comment by DaxDupont** [comment]
Plague Inc SS13 edition

**Comment by davethwave** [comment]
This sounds like a great addition to the ghost event spawn

**Comment by ExcessiveUseOfCobblestone** [comment]
@Cruix I like the kill viro obj but we've had "friendly side antags that give epic heals with a survive obj" (borers) before and they're no longer in the game, I'd hate for this feature to go the same route.

**Comment by Cruix** [comment]
Borers were removed because they were buggy and no one wanted to fix them. In any case, healing symptoms take a lot of time to unlock, and in order to reach them you will have to get some of the more annoying symptoms first in order to spread.

**Comment by epochayur** [comment]
this is seriously cool

**Comment by XDTM** [comment]
This could definitely use a set of exclusive symptoms. 
Biosuit piercing would prevent people in hardsuits and biosuits from shrugging off everything.
Wandering spores that behave like spiderlings could infect targets through vents, and people would weld vents during a contagion.
Some late-stage symptom that would allow the contagion to brainwash a vulnerable target, maybe defined as someone who took enough brain damage.
Something that can act as an offensive measure against virus immune targets, like toxin emission.
Maybe a zombie symptom as an endgame thing.

**Comment by CosmicScientist** [comment]
>we plague inc now

Does no-one remember pandemic?

I require Madagascar to be a part of this PR as its balance fits with the rest of the game.

**Comment by davethwave** [comment]
On the top of my head I can think of perhaps a few things missing that could be good from the symptom list. Such as anything that could cause brain damage, perhaps investing points that could buff the virus marginally in one way such as stealth +1. Trauma inducing diseases, perhaps a medical phobia symptom, and maybe since this kind of a reference to plague inc see if there might be something interesting there that could be implemented.

**Comment by Kromgar** [comment]
PAPA NURGLE GiVES YOU HIS BLESSING 

image

**Comment by ShadowDeath6** [comment]
@cosmicscientist

tbf plague inc is much more mainstream because it's not a flash game, and it has the same basic format as pandemic considering it's mentioned that it took a ton of inspiration from it

I wouldn't be opposed to a space madagascar space ruin that spawned the disease if the person entered the ruin

also :+1: I've been waiting for this since 2015

**Comment by Kromgar** [comment]
@ShadowDeath6 

Space Madagascar ruin that's a closed harbor

**Comment by Doppeltongue** [comment]
"CMO, an assistant in tool storage is coughing!"
"SHUT. DOWN. EVERYTHING."

I'm assuming that this antag would always/sometimes replace the current Level 7 biohazard event?

**Comment by davethwave** [comment]
deadspace!!!

**Comment by Cruix** [comment]
coughing

**Comment by MMMiracles** [comment]
You need a way to fight vaccines, as there really isn't a way to prevent people from just getting a vaccine as soon as on is available. You're pretty much dead once they start spreading around unless you get real lucky and a single person doesn't go for once since you're not intrusive enough, but even then you're stuck on a single person surrounded by immune people who'll just tell the last host to grab a vaccine. Having something like a way to re-adapt completely under a new strain with a fresh start would be a good thing to have when you need to avoid being completely eradicated.

There should probably be a way to tell the difference between potential infection targets and immune targets, so you don't end up blowing your cover of the potentially unknowing infected by sneezing on an already-immune target. 

Need a way to refresh the infection menu, as it doesn't update with stuff like newly available symptoms to purchase or infected individuals. Having to open/close the menu every time for an updated view is kinda annoying.

**Comment by PKPenguin321** [comment]
I really hope this is as awesome as I'm imagining it

File: code/datums/action.dm
Comment (AnturK) [inline code/datums/action.dm]: Should probably also stop updating when it has no owner.

File: code/datums/diseases/_disease.dm
Comment (AnturK) [inline code/datums/diseases/_disease.dm]: VV lock this just in case.

File: code/datums/diseases/advance/advance.dm
Comment (AnturK) [inline code/datums/diseases/advance/advance.dm]: Is there any reason to have /advance subtype now ? As in the core diseases being static was the major difference from what i can remember.

File: code/modules/admin/admin_verbs.dm
Comment (AnturK) [inline code/modules/admin/admin_verbs.dm]: Could also change the mob list

File: code/modules/antagonists/disease/disease_mob.dm
Comment (AnturK) [inline code/modules/antagonists/disease/disease_mob.dm]: Listening to Moved signal would probably be better for this.

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks pretty good, just some minor stuff

**Comment by drawsstuff** [comment]
Maybe a early, cheap, harmful symptom and beneficial symptom to let people know what tree and path you plan so a cough doesnt send everyone into a panic thinking you're gonna murderbone. Alternatively, trick people into thinking your beneficial, then replace your symptoms to be harmful from that point on out!

**Comment by Ratvar** [comment]
add  fart as reverse lowrange sneeze

File: code/datums/diseases/_disease.dm
Comment (Cruix) [reply code/datums/diseases/_disease.dm]: You cannot access static proc vars from VV.

File: code/datums/diseases/advance/advance.dm
Comment (Cruix) [reply code/datums/diseases/advance/advance.dm]: The /advance subtype has datum symptoms, while regular diseases do not.

**Comment by Cruix** [comment]
Everything else I can do.

I am not going to add new symptoms, vaccine readaptation, or the like until I know that the premise of the disease is working.

File: code/datums/diseases/_disease.dm
Comment (AnturK) [reply code/datums/diseases/_disease.dm]: Oh right, its local, carry on.

**Comment by Nabski89** [comment]
I look forward to being lynched because people think I am responsible for this.

**Comment by KorPhaeron** [comment]
Fantastic

**Comment by MoreRobustThanYou** [comment]
add a virologist traitor item that summons one

**Comment by XDTM** [comment]
Don't

**Comment by MoreRobustThanYou** [comment]
make it 18 TC, so the cost of a holopara

**Comment by XDTM** [comment]
no, if traitors can buy it it'll get stale really fast, get burnt out, and plus it will be associated with "lynch viro" when the viro is the one you should turn to when the contagion appears

**Comment by MoreRobustThanYou** [comment]
oh

**Comment by ExcessiveUseOfCobblestone** [comment]
hi mrty!

**Comment by MoreRobustThanYou** [comment]
i regularly leave input on PRs, don't act like seeing me is rare.

**Comment by tortellinitony** [comment]
Try to make it use TGUI

**Comment by davethwave** [comment]
Can't wait until this is test merged

**Comment by ChuckTheSheep** [comment]
Threw together some nicer graphics: https://file.io/8TDphb . I like this idea, good luck!

**Comment by Cruix** [comment]
Thanks a lot for making my codersprites look even more terrible than they already did.
I'm going to leave the virus_infected one the same because I am not a huge fan of the gas cloud, and because I want it to be white and colored with the color var so there can be multiple different colors for different viruses.

**Comment by ChuckTheSheep** [comment]
Thank you, and no worries, use which you prefer. :)

**Comment by Buggy123** [comment]
>Does no-one remember pandemic?

It actually took me a moment to remember what the popular version of that game was, I could only remember Pandemic at first. So, yes, but Pandemic is no longer culturally significant so I have no reason to reference it.

**Comment by Ordonis** [comment]
Seeing this combined with some of the recent updates to surgery/traits could make this extra entertaining

File: code/modules/antagonists/disease/disease_mob.dm
Comment (AnturK) [inline code/modules/antagonists/disease/disease_mob.dm]: I feel this would make more sense on key focus level but that can be changed later.

**Comment by davethwave** [comment]
didn't notice this got merged, can't wait to see it in action!

**Pull request merged**
